Lazy/Lazy.Src/ILazy.cs

```diff
@@ -0,0 +1,15 @@
+namespace Lazy;
+
+/// <summary>
+/// An interface for realisation of class for lazy calculations.
+/// </summary>
+/// <typeparam name="T">Return value of function.</typeparam>
+public interface ILazy<T>

```

Comment:
```suggestion
public interface ILazy<out T>
```
Если генерик только возвращает или только принимает значение параметра-типа, то стоит указывать вариантность, это может сильно упростить его использование.